### PR TITLE
fix parameter handling

### PR DESCRIPTION
**Pull request by cjwooo**

### CODE REVIEW

File: parallel/integer.cu [hunk 1]
```diff
@@ -64,9 +64,9 @@ static void packBits(integer output, int* outputIndex_orig, int* bitIndex_orig,
 // Read an integer from a stream. Parse nine characters at a time, from right to
 // left, packing bits accordingly. Nine characters because it's guaranteed to
 // fit in a single int32_t.
-void integer_fread(integer output, char** lineptr, size_t* n, FILE* stream) {
-   memset(output, '\0', sizeof(output));
-   ssize_t numBytes = getline_(line, n, stream);
+void integer_fread(integer *output, char** lineptr, size_t* n, FILE* stream) {
+   memset(*output, '\0', sizeof(*output));
```
Comment (mitchellwrosen) [inline parallel/integer.cu]: Why *output? it's a int32_t[32], so dereferencing is just one int32_t

File: parallel/integer.cu [hunk 2]
```diff
@@ -81,10 +81,10 @@ void integer_fread(integer output, char** lineptr, size_t* n, FILE* stream) {
       if (lineIndex < 0)
          lineIndex = 0;
 
-      int32_t segment = atoi(line + lineIndex);
-      packBits(output, &outputIndex, &bitIndex, segment);
+      int32_t segment = atoi(*lineptr + lineIndex);
+      packBits(*output, &outputIndex, &bitIndex, segment);
```
Comment (mitchellwrosen) [inline parallel/integer.cu]: again, should be output, not *output, right?

File: parallel/integer.cu [hunk 3]
```diff
@@ -95,9 +95,9 @@ __device__ void integer_copy(integer output, const integer input) {
       output[i] = input[i];
 }
 
-__device__ void integer_fromInt(integer output, const int32_t input) {
-   memset(output, '\0', sizeof(output) - sizeof(int32_t)); // Zero all but lowest order byte
-   output[N-2] = input;
+__device__ void integer_fromInt(integer *output, const int32_t input) {
```
Comment (mitchellwrosen) [inline parallel/integer.cu]: parameter should be "integer output"

File: parallel/integer.cu [hunk 4]
```diff
@@ -95,9 +95,9 @@ __device__ void integer_copy(integer output, const integer input) {
...
+   memset(*output, '\0', sizeof(*output) - sizeof(int32_t)); // Zero all but lowest order byte
```
Comment (mitchellwrosen) [inline parallel/integer.cu]: no dereference

File: parallel/integer.cu [hunk 5]
```diff
@@ -95,9 +95,9 @@ __device__ void integer_copy(integer output, const integer input) {
...
+   (*output)[N-2] = input;
```
Comment (mitchellwrosen) [inline parallel/integer.cu]: no dereference

**Comment by cjwooo** [comment]
But if you just pass integer to a function, it's by reference so output is just a pointer, so sizeof(output) would be sizeof(int32_t *) and memset(output) would overwrite the pointer, right?

**Comment by mitchellwrosen** [comment]
No, because it's an int32_t[32], not an int32_t*, so sizeof(integer) is 4 \* 32

**Comment by cjwooo** [comment]
Ok, just updated.

**Pull request merged**